app/screens/image_preview/video_preview.js

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    StyleSheet,
+    View
+} from 'react-native';
+import Video from 'react-native-video';
+import RNFetchBlob from 'react-native-fetch-blob';
+
+import EventEmitter from 'mattermost-redux/utils/event_emitter';
+
+import VideoControls, {PLAYER_STATE} from 'app/components/video_controls';
+import {DeviceTypes} from 'app/constants/';
+
+import Downloader from './downloader.ios';
+
+const {VIDEOS_PATH} = DeviceTypes;
+
+export default class VideoPreview extends PureComponent {
+    static propTypes = {
+        deviceHeight: PropTypes.number.isRequired,
+        deviceWidth: PropTypes.number.isRequired,
+        file: PropTypes.object,
+        onFullScreen: PropTypes.func,
+        onSeeking: PropTypes.func,
```

Comment:
If we don't have defaults might not need to mark them as required just for debugging sake

Reply:
Yeah, I think we want to default to marking props as required unless we have a good reason for them not to be